### PR TITLE
Scheduled push for devcontainer image

### LINKED ISSUE #318: Java Docker environment for users that can't meet current Helium requirements.
## Description

Give users a Docker environment for setting up Helium. Some users can't meet the current requirements, but they are able to use Docker Desktop. There is an existing Java Docker image that is being used in codespaces, and Github Actions to push new images when merging to main. The next step is to add is a recurring build and push action to get package updates.

- [x] A Dockerfile for helium java that can be used locally and in codespaces. This should include all the tools needed to step through the Helium setup.
- [x] Build the Dockerfile as part of ci/cd. Build on changes to the file, build and push to Docker Hub when merging to main.
- [ ] Build and push the Dockerfile on a recurring schedule for package updates.
- [x] For codespaces, use the pre-built images for faster startup.

- References https://github.com/retaildevcrews/helium/issues/579

### PR DESCRIPTION
**Pull request by sivamu**
# Type of PR

- [ ] Documentation changes
- [ ] Code changes
- [ ] Test changes
- [x] CI-CD changes
- [ ] GitHub Template changes

## Purpose of PR
Push the java devcontainer Dockerfile on a recurring schedule for package updates.

## Validation

- [ ] Unit tests updated and ran successfully
- [ ] Update documentation or issue referenced above

## Issues Closed or Referenced

- Closes #318

### CODE REVIEW

File: .github/workflows/codespaces.yml [hunk 1]
```diff
@@ -1,6 +1,9 @@
 name: Java Codespaces Image Build
 
 on:
+  schedule:
```
Comment (bartr) [inline .github/workflows/codespaces.yml:4]: can we change this to 0 3 * * 0  to run weekly?

**Review by bartr — APPROVED** [review]
approved with a suggestion

File: .github/workflows/codespaces.yml [hunk 2]
```diff
@@ -1,6 +1,9 @@
...
+    # Run weekly at 3:00 AM
+    - cron: "0 3 * * 0"
   push:
     branches:
       - master
```
Comment (AAkindele) [inline .github/workflows/codespaces.yml:9]: 🤔 💭 should this be `main` now, or does this still work?

Comment (sivamu) [reply .github/workflows/codespaces.yml:9]: d'oh, good point i keep forgetting this.

**Review by AAkindele — APPROVED** [review]

**Pull request merged**